en/middleware/micrortps.md

```diff
@@ -30,58 +32,84 @@ The main elements of the architecture are the client and agent processes shown i
 
-For *automatic code generation* (via the normal PX4 firmware build process) this set must be listed in the **/path/to/PX4/Firmware/src/modules/micrortps_bridge/CMakeLists.txt** file.
+The `px4_ros_com` package build structure is structured in a way that, when one builds the package, it automatically generates all the required components of the bridge that allow the developer to use the generated messages on its own nodes, being it on the ROS2 or in the ROS side.
 
-```cmake
-set(config_rtps_send_topics
-  sensor_combined
-   # Add new topic...
-   )
+## Supported uORB messages
 
-set(config_rtps_receive_topics
-   sensor_baro
-   # Add new topic...
-   )
+The generated bridge code will enable a specified subset of uORB topics to be published/subscribed via RTPS. This applicable on both ROS or non-ROS applications.
+
+For *automatic code generation* (via the normal PX4 Firmware build process or, in the case of the `px4_ros_com`, through the build process of the package), there's a yaml definition file on the PX4 Firmware directory called `uorb_rtps_message_ids.yaml`, found in **/path/to/PX4/Firmware/msg/tools/** dir, where all the uORB messages (RTPS) ID's are set and where is also set which messages should be set to be sent, received, or both, in the RTPS stream.
+
+> **Caution** All new uORB messages which one adds to the Firmware side should have a respective RTPS ID set when one wants the data to be published and/or subscribed in the RTPS stream.
+
+```yaml
+rtps:
+  - msg: actuator_armed
+    id: 0
+  - msg: actuator_control
+    id: 1
+  - ...
+  - msg: airspeed
+    id: 5
+    send: true
+  - msg: battery_status
+    id: 6
+    send: true
+  - msg: camera_capture
+    id: 7
+  - msg: camera_trigger
+    id: 8
+    receive: true
 ```
 
-> **Caution** At time of writing (August 2017), only the small set of uORB topics listed above are included in our cmake files: **posix_sitl_default.cmake**, **nuttx_px4fmu-v4_default.cmake**, **posix_sdflight_default.cmake**. It is likely you will need to edit your *cmake* file and add additional uORB topics. In future we hope to define a larger standard set. 
+In the case of `px4_ros_com`, the `uorb_rtps_message_ids.yaml` is transformed in a way that the message names become PascalCased and there's a swap between the received and the sent messages. The logic behind this is that if a message is sent from the client side, then it's received on the agent side, and vice-versa. The naming of the message is indifferent on the client-agent communication, but it is critical on the ROS2 side, since the message naming has to follow the the PascalCase convention, which means that the generated IDL files also need to follow this same convention.
```

Comment:
This information doesn't seem to match the topic "## Supported uORB messages".
- It is additional/supporting so perhaps should be in  a note, a different section or a sub section.
- Does this transformation only happen when you build using `px4_ros_com`? Or are the ROS IDl files built whenever you build firmware. 

I'll suggest best path once you tell me the answer to the second point. Something like (if this is correct!)

```
The yaml file is used to create IDL files for specified uORB messages as part of the build process.

Separate IDL files are build for use by ROS2 nodes and other RTPS nodes. 

> **Note** ROS2 requires PascalCased node names. The `px4_ros_com` build process automatically creates appropriate IDL files for use with ROS2 nodes. 
  These are mapped by the agent to the equivalent messages PX4-side, using the RTPS ID.
```

Reply:
> It is additional/supporting so perhaps should be in a note, a different section or a sub section.

Yeah I should add it as a note.

> Does this transformation only happen when you build using `px4_ros_com`? Or are the ROS IDl files built whenever you build firmware.

This just happens when you build the `px4_ros_com` package.

Actually this needs to be clarified in a different matter. The PX4 Firmware only generates the IDL files required for a bridge between the client-agent and a standard listener app. For the ROS2 side, it's actually the `rosidl_generator` that takes care of that, not PX4. I will make this clear.